File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    private static StringTokenizer st;
    private static int total;
    private static int A;
    private static int B;
    private static int C;
    private static int maxVal;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        total = 0;
        do {
            st = new StringTokenizer(br.readLine());
            A = Integer.parseInt(st.nextToken());
            B = Integer.parseInt(st.nextToken());
            C = Integer.parseInt(st.nextToken());

            total = A + B + C;
            if (total == 0) break;

            maxVal = Math.max(Math.max(A, B), C);
            if (maxVal < total - maxVal) {
                if (A == B && A == C) {
                    sb.append("Equilateral");
                } else if (A == B || A == C || B == C) {
                    sb.append("Isosceles");
                } else {
                    sb.append("Scalene");
                }
            } else {
                sb.append("Invalid");
            }
            sb.append("\n");
        } while (true);
        System.out.print(sb);
    }
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys
input = sys.stdin.readline

def solution():
    while True:
        answer = "Invalid"

        A, B, C = map(int, input().strip().split())

        total = A + B + C
        if total == 0:
            break
        max_val = max(A, B, C)

        if max_val < total - max_val:
            if A == B == C:
                answer = "Equilateral"
            elif A == B or A == C or B == C:
                answer = "Isosceles"
            else:
                answer = "Scalene"
        print(answer)


solution()